pkg/recoveryhandler/zap.go

```diff
@@ -30,7 +30,7 @@ type zapRecoveryWrapper struct {
 // Println logs an error message with the given fields
 func (z zapRecoveryWrapper) Println(fields ...interface{}) {
 	// if you think i'm going to check the type of each of the fields and then logger with fields, you're crazy.
-	z.logger.Error(fmt.Sprintln(fields))
+	z.logger.Error(fmt.Sprintln([]interface{}(fields)))
```

Comment:
don't we want `fields...` instead here?

Reply:
Well, the existing test case won't work out that way. It expects `Println(1, 2, 3)` to log as `message: [1, 2, 3]`, not `message: 1, 2, 3`. I'm just wondering why one of the `zap` convenience wrappers weren't used here. Might replace.

Reply:
looking at gorilla's interface:
https://github.com/gorilla/handlers/blob/7e0847f9db758cdebd26c149d0ae9d5d0b9c98ce/recovery.go#L10

I don't think it implies that if you call `logger(1, 2, 3)` that you would get `[1, 2, 3]` output. Rather, it should be equivalent to calling `println(1, 2, 3)`, which will print `1 2 3`.

Reply:
basically, our implementation of the adapter was incorrect